### ISSUE TITLE
Associative memory never resets an evicted row, so a freshly allocated key inherits the evicted key's values, utility, and counts (small-table ablations score below chance)

### ISSUE THREAD

**Issue by Prabhat1308** [SlopDotCash/asi#456]
## Summary

`AssociativeMemoryLearner.update` (`alberta_framework/core/associative_memory.py`) documents that "the table budget is reused by **replacing** low-utility rows" and that "row values are decayed and updated toward the observed next label". In `row_step`, when a key is not found and a slot is reused by eviction, only `keys[slot]` is overwritten: `old_row = carry.values[slot]`, `old_utility = carry.utility[slot]`, and `counts.at[slot].add(1.0)` are carried over from the **previous occupant**. The row therefore no longer represents its key: a freshly allocated key inherits the evicted key's values, utility, and counts, and predicts the previous occupant's label at high confidence until enough of its own writes accumulate.

Because `AlbertaPipeline` uses `associative.predict(...).probabilities` as the Step 2 feature vector feeding Steps 3/4, and `run_step2_associative_smoke` reports `final_window_nll` off the same path, any budget ablation below the default `max_features=512` (which records 0 replacements) measures inheritance, not capacity — a fresh key emits the previous occupant's label at p≈1, so the reported NLL/accuracy for a smaller table is systematically anti-predictive rather than merely degraded.

## Falsifier (deterministic, pristine `main` @ `b8a37de`)

Isolated: `vocab_size=4, block_size=2, suffix_length=2, feature_family="position_token", max_features=2`; write context A with label 3 thirty times, then write context B with label 1 once (evicting A):

```
B seen once, label=1.  found = [1, 1, 0]
  values for B's rows = [[0.0, 1.0, 0.0, 3.995], ...]   <- 3.995 at class 3 (A's label) vs 1.0 at class 1 (B's only write)
  logits argmax = 3
  B's key only starts predicting its own label after 4 total writes
```

Steady state on a periodic 4-class stream (`repro_evict_stream.py`, 320 replacements at small tables):

```
max_features=  2  steady-state acc=0.000  loss=1.406  (chance acc=0.25, uniform loss=1.386)  replacements=318
max_features=  4  steady-state acc=0.000  loss=1.406  replacements=316
max_features=  8  steady-state acc=1.000  loss=0.000  replacements=0
```

Accuracy 0.0 — below chance — because every fresh key emits the evicted key's label. Expected: a slot reused for a new key starts from an empty row, zero utility, and a count of one (with the fix the same stream gives `acc 0.25 / 0.50 / 1.00` for `max_features 2 / 4 / 8`).

## Dedup

Open #446 / PR #449 (deepanshu-yd) harden `AssociativeMemoryConfig` scalar/int validation — a different surface in the same module (`_validate_config`, config class); merge-tree composes cleanly. Adjacent closed: PRs #196, #90 (associative-memory 0×inf / inf−inf guards). `gh issue/pr list --search "eviction" / "replacement row"`: #432 (ExperientialMemoryPolicy, different module), #407/#371 (UPGD/CBP recycling budgets, unrelated). No issue covers the eviction row reset.

Fix PR follows (failing-test-first).
